kratos/geometries/geometry_shape_function_container.h

```diff
@@ -237,6 +304,28 @@ class GeometryShapeFunctionContainer
         return mShapeFunctionsLocalGradients[ThisMethod][IntegrationPointIndex];
     }
 
+    /*
+    * @brief access each item separateley.
+    * @param DerivativeOrderIndex defines the wanted order of the derivative
+    * @param DerivativeOrderRowIndex within each derivative the entries can
+    *        be accessed differently.
+    * @return the shape function or derivative value related to the input parameters.
```

Comment:
could u put here an example of the intended use of this?

Reply:
This is just one way to use this container

Reply:
My suggested way will be:
const Matrix& ShapeFunctionDerivatives(IndexType Order, IndexType IntegrationPointIndex)
Possible for orders: 1-...
Due to backwards compatibility 0 is not possible

Reply:
what i was asking is how the data is actually stored in the return matrix (essentially i am asking for a small documentation of how to use it in this text

Reply:
I'll add this function